File: tests/fixtures/rust/sample.rs
// Rust fixture: free function + struct with impl block methods

pub fn helper(x: i32) -> i32 {
    x * 2
}

pub fn top_level_one(x: i32) -> i32 {
    helper(x)
}

pub struct Greeter {
    pub name: String,
}

impl Greeter {
    pub fn new(name: String) -> Self {
        Greeter { name }
    }

    pub fn greet(&self) -> i32 {
        helper(self.name.len() as i32)
    }
}

File: tests/fixtures/php/sample.php
<?php

function helper($x) {
    return $x * 2;
}

function topLevelOne($x) {
    return helper($x);
}

class Greeter {
    private $name;

    public function __construct($name) {
        $this->name = $name;
    }

    public function greet() {
        return helper(strlen($this->name));
    }
}
